### PR TITLE
Add support for signing Windows binaries without a destination path

### PR DESCRIPTION
**Pull request by fheinecke**
This allows callers of the `Invoke-SignBinary` powershell commandlet to pass only the path to the unsigned binary. In this case, the signed binary will overwrite the unsigned binary. This is needed so that Electron builder can be "tricked" into using our signing code when signing Teleport Connect on Windows.

Needed for https://github.com/gravitational/teleport/pull/41963.

### CODE REVIEW

File: build.assets/windows/build.ps1 [hunk 1]
```diff
@@ -288,18 +288,32 @@ function Invoke-SignBinary {
...
 
+    if (! $SignedBinaryPath) {
+        $ShouldMoveSignedBinary = true
+        $SignedBinaryPath = Join-Path -Path $(New-TempDirectory) -ChildPath "signed.exe"
+    }
+
     Write-Host "Signing $UnsignedBinaryPath using WSL sign-binary script:"
     wsl-ubuntu-command sign-binary "$UnsignedBinaryPath" "$SignedBinaryPath"
+
+    if ($ShouldMoveSignedBinary) {
```
Comment (camscale) [inline build.assets/windows/build.ps1:314]: This var is uninitialised if `$SignedBinaryPath` is provided as a parameter. Does this matter? Should it be initialised to `false` before setting it to true conditionally above?

Comment (fheinecke) [reply build.assets/windows/build.ps1:314]: No it's fine in powershell. Undefined values are considered null, which evaluates to false. Here's an example:

```powershell
Write-Host $SomeInvalidVar # Nothing is logged

if ($SomeMadeUpVar) {
  Write-Host "If this is logged then the PR logic is flawed"
} else {
  Write-Host "Undefined values are null, which is falsey" # This is printed
}
```

**Review by camscale — APPROVED** [review]

**Review by rosstimothy — COMMENTED** [review]
Were there any tag builds performed from this branch to verify this works as expected?

**Comment by fheinecke** [comment]
@rosstimothy Not yet, as it's pretty low risk. I'll start one now though

**Review by r0mant — APPROVED** [review]
lgtm if the test build succeeds

**Comment by fheinecke** [comment]
Test build succeeded for windows, but Linux failed due to something completely unrelated (and can be ignored)

**Pull request merged**